extensions/amp-story/1.0/amp-story.js

```diff
@@ -963,6 +964,16 @@ export class AmpStory extends AMP.BaseElement {
       .then(() => this.whenPagesLoaded_(PAGE_LOAD_TIMEOUT_MS))
       .then(() => this.markStoryAsLoaded_());
 
+    // Story is being prerendered: resolve the layoutCallback when the first
+    // page is built. Other pages will only build if the document becomes
+    // visible.
+    if (!Services.viewerForDoc(this.element).hasBeenVisible()) {
+      return whenUpgradedToCustomElement(firstPageEl).then(() =>
+        firstPageEl.whenBuilt()
```

Comment:
Doesn't this need to be chained off of at least some of the stuff above?  Like, if we short circuit here, then the user clicks into the story before the other promises have resolved, won't they enter in a weird state?

(For example: consent not initialized, progress bar not built, etc.)

Reply:
This is actually what's already happening today, this fix only changes when the `layoutCallback` resolves so the story is displayed when the first page is built, and not when the render service times out.
Everything you mentioned runs once `initializePages_()` resolves, when all the pages were built. Technically, when the document becomes active.

We could re-think what should build or not in prerendering vs normal rendering, but it'd be a bigger refactoring. Basically when the `initializePages_()` resolves, we build the system layer (would need to know how many pages we have), start playing the first page (def don't want to do that), etc.